src/tema/hour/TimeDisplay.java

```diff
@@ -0,0 +1,127 @@
+package tema.hour;
+
+import java.util.Scanner;
+
+/**
+ * the class  gets the imput for hour and minutes from user and returns the time based on his imput
+ * @author Marius Cristea
+ * @version 1
+ */
+public class TimeDisplay {
+    /**
+     * displays the time with the hour and minutes received from the user
+     * validates the values making sure the hour is betweem 0 and 23 and the minutes between 0 and 59
+     * the format is HH:MM
+     * @param args
+     */
+    public static void main(String[] args) {
+        int hour, minutes;
+        hour = getNumber("hour");
+        minutes = getNumber("minutes");
+        System.out.println("The time is: "+ modifyFormat(Integer.toString(hour)) + ":" + modifyFormat(Integer.toString(minutes)));
+    }
+
+    /**
+     * modifies the string is the lenght is 1 by adding in front a 0
+     * @param a
+     * @return a string with the first character 0 if the length of the string is 1 else returns the same string
+     */
+    private static String modifyFormat(String a){
+        if(a.length() == 1){
+            return 0+a;
+        }
+        return a;
+    }
+
+    /**
+     * returns an integer value inserted by the customer based on the string value may return hour or minutes
+     * @param a counter in order to request the correct number from customer
+     * @return interger value for hour or minutes inserted by customer
+     */
+    private static int getNumber(String a){
+        int finalNumber;
+        String number;
+        int j = 0;
+        Scanner keyboard = new Scanner(System.in);
+        if (a == "hour"){
+            do {
+                if (j == 0) {
+                    System.out.println("Please insert the " + a);
```

Comment:
if you did "if a == hour" why bother with "please insert the" + a.. it would have been simpler to just write the complete message.